src/fides/api/service/connectors/sql_connector.py

```diff
@@ -510,6 +511,19 @@ def query_config(self, node: ExecutionNode) -> BigQueryQueryConfig:
         """Query wrapper corresponding to the input execution_node."""
         return BigQueryQueryConfig(node)
 
+    # Overrides SQLConnector.test_connection
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        try:
+            bq_schema = BigQuerySchema(**self.configuration.secrets)
+            client = bq_schema.get_client()
+            all_projects = [project for project in client.list_projects()]
+            if all_projects:
+                return ConnectionTestStatus.succeeded
+            else:
+                raise ConnectionException("No datasets found with the provided credentials.")
+        except Exception as e:
+            raise ConnectionException(f"Connection error: {e}")
```

Comment:
this is solid for the exception that's raised up (and hopefully gets to the UI?) in addition, it might be nice to _log_ the exception here too, ideally with a full stack trace - since this endpoint is specifically for troubleshooting, giving as much error information as possible can be helpful.

you can use `logger.exception` - i think that may give you a stack trace, or you may need to still specify some args/options to get a full stack trace.

Reply:
logger exception here: https://github.com/ethyca/fides/pull/5138/commits/16d65db4c965c945833221f7d03c3fb705a31073#diff-15895ed412c26a8f5c097a7eddfa11273da29a8d999e7c342bdeb475224fff8aR537 I do see the exception in logs if I induce one